config/src/errors.rs

```diff
@@ -0,0 +1,180 @@
+//! Errors for graph-sdk crate
+//!
+
+use thiserror::Error;
+
+pub type DsnpGraphResult<T> = std::result::Result<T, DsnpGraphError>;
+
+#[derive(Debug, Error)]
+pub enum DsnpGraphError {
```

Comment:
Do we want to `#[repr(C, u8)]` this? Or otherwise we'll need to wrap it in something for the FFI.